zerver/migrations/0173_user_message_add_and_index_is_private_flag.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-06-14 13:39
+from __future__ import unicode_literals
+
+import sys
+import bitfield.models
+from django.db import migrations, transaction
+from zerver.lib.migrate import create_index_if_not_exist  # nolint
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import F
+
+def reset_is_private_flag(
+        apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserMessage = apps.get_model("zerver", "UserMessage")
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    user_profile_ids = UserProfile.objects.all().order_by("id").values_list("id", flat=True)
+
+    i = 1
+    total = len(user_profile_ids)
+    print("\nStart resetting is_private flag...")
+    for user_id in user_profile_ids:
+        if (i % 50 == 1):
+            print("%s/%s" % (user_id, total))
+            sys.stdout.flush()
+        flag_set_objects = UserMessage.objects.filter(flags=UserMessage.flags.is_private,
+                                                      user_profile__id = user_id)
+
+        with transaction.atomic():
+            # We only need to do this because a previous migration didn't clean the field.
+            flag_set_objects.update(flags=F('flags').bitand(~UserMessage.flags.is_private))
+        if (i % 50 == 1):
+            percent = round((user_id / total) * 100, 2)
+            print("Processed %s/%s %s%%" % (user_id, total, percent))
```

Comment:
Why do you have two print statements in the loop, and only one of them doing the flush?  I think this needs more thought; it also seems like we print both reports the first time around after just a single row.